lib/service/service.go

```diff
@@ -1401,12 +1404,32 @@ func (process *TeleportProcess) onExit(serviceName string, callback func(interfa
 	})
 }
 
+// WebSessions returns a new web session manager
+// Implements types.WebSessionsGetter
+func (r cacheServices) WebSessions() types.WebSessionInterface {
+	return r.Services.WebSessions()
+}
+
+// WebTokens returns a new web token manager
+// Implements types.WebTokensGetter
+func (r cacheServices) WebTokens() types.WebTokenInterface {
+	return r.Services.WebTokens()
+}
+
+// cacheServices is a wrapper for auth.Services that provides
+// services.Services interface
+type cacheServices struct {
```

Comment:
This wrapper allows the server's side of the services to be used in the cache. @awly I know you asked the question as to why this is needed and why cache is not using client-facing interfaces - I will look into that but off the top of my head, I think it makes the split between the two and the collection management uses server-side APIs (e.g. Upserts that were previously stubbed with no-op implementations on the client side) hence this is currently required.